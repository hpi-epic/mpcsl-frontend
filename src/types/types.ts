import { IAPIGraphEdges, IAPIGraphNode } from './graphTypes';

export interface IIDClass {
  id: number;
}

export interface ICreateDatasetGenerationJob extends IIDClass {
  datasetName: string;
  kubernetesNode?: string;
  nodes: number;
  samples: number;
  edgeProbability: number;
  edgeValueLowerBound: number;
  edgeValueUpperBound: number;
}

export interface IObservationMatrix extends IIDClass {
  load_query: string;
  name: string;
  description?: string;
  data_source?: string;
  time_created?: string;
}

export interface IObservationMatrixMetadata {
  variables: number;
  time_created: number;
  observations: number;
  data_source: string;
  query: string;
  has_ground_truth: boolean;
}

export type JobStatus = 'running' | 'done' | 'error' | 'cancelled' | 'waiting';

export enum JobErrorCode {
  UNSCHEDULABLE = -1,
  IMAGE_NOT_FOUND = -2,
  UNKNOWN = -127
}

export enum BadgeStatus {
  running = 'processing',
  waiting = 'processing',
  done = 'success',
  error = 'error',
  cancelled = 'warning'
}

export interface IExperiment extends IIDClass {
  dataset_id: number;
  dataset?: number;
  name: string;
  description?: string;
  execution_time_statistics?: { [name: string]: number };
  algorithm_id: number;
  parameters: IParameters;
  last_job?: IJob;
}

export interface ICreateExperiment {
  dataset_id: number;
  name: string;
  description: string;
  algorithm_id: number;
  parameters: { [name: string]: any };
}

interface IRequiredParameter {
  required?: boolean;
}

export interface INumberParameter extends IRequiredParameter {
  type: 'int' | 'float';
  minimum?: number;
  maximum?: number;
  default?: number;
}

export interface IEnumParameter extends IRequiredParameter {
  type: 'enum';
  values: string[];
  default?: string;
}

export interface IStrParameter extends IRequiredParameter {
  type: 'str';
  value: string;
  default?: string;
}

export type IParameter = INumberParameter | IEnumParameter | IStrParameter;

export type IParameters = { [name: string]: IParameter };

export interface IAlgorithm extends IIDClass {
  package: string;
  function: string;
  script_filename: string;
  backend: string;
  description: string;
  valid_parameters: IParameters;
  needs_gpu?: boolean;
}

export interface IErrorType {
  rate: number;
  edges: number[][];
}

export interface IJob extends IIDClass {
  experiment_id: number;
  start_time: string;
  end_time: string;
  status: JobStatus;
  pid?: number;
  result?: {
    id: number;
    job_id: number;
    start_time: string;
    end_time: string;
    execution_time: number;
    dataset_loading_time: number;
    meta_results: any;
  };
}

export interface IComparisonStatistics {
  graph_edit_distance: number;
  mean_jaccard_coefficient: number;
  error_types: {
    false_positives: IErrorType;
    true_positives: IErrorType;
    false_negatives: IErrorType;
    true_negatives: IErrorType;
  };
}

interface IAPIDistributionBase {
  dataset: {
    time_created: string;
    id: number;
    description?: string;
    name: string;
    data_source?: string;
    load_query?: string;
  };
  node: {
    result: number;
    name: string;
    result_id: number;
    id: number;
  };
}

export interface IAPIDistributionContinous extends IAPIDistributionBase {
  categorical: false;
  bin_edges: number[];
  bins: number[];
}

interface ICategoricalBin {
  [bin: string]: number;
}

export interface IAPIDistributionCategorical extends IAPIDistributionBase {
  categorical: true;
  bins: ICategoricalBin;
}

export type IAPIDistribution =
  | IAPIDistributionCategorical
  | IAPIDistributionContinous;

export interface IAPINodeContext {
  edges: IAPIGraphEdges[];
  context_nodes: IAPIGraphNode[];
  main_node: IAPIGraphNode;
}

export type IAPIAllNodesContext = {
  node_contexts: IAPINodeContext[];
};

export enum GraphExportFormat {
  GEXF = 'GEXF',
  GraphML = 'GraphML',
  GML = 'GML'
}

export interface IStateJobsManagement {
  modalVisible: boolean;
  downloadModalVisible: boolean;
  extraLines: number; // necessary for linux support
  currentJobId?: number;
  jobList: IJob[];
  experiment?: IExperiment;
  format: GraphExportFormat;
  currentResultID?: number;
}

export interface IMatchParams {
  experimentId: string;
  datasetId: string;
}

export type IExperimentJobs = {
  jobs: IJob[];
} & IExperiment;

export interface IAPIConfounders {
  confounders: string[][];
}

export interface IDatasetGenerationJob extends IJob {
  dataset_id: number;
  datasetName: string;
  nodes: number;
  samples: number;
  edgeProbability: number;
  edgeValueLowerBound: number;
  edgeValueUpperBound: number;
}
