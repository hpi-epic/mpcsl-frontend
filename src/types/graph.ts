type NodeID = string | number;

export interface D3Graph {
  nodes: Array<D3GraphNode>;
  links: Array<D3GraphLink>;
}

export interface D3GraphNode {
  id: NodeID;
  x?: number;
  y?: number;
  fx?: number;
  fy?: number;
  vx?: number;
  vy?: number;
}

export interface D3GraphLink {
  source: NodeID;
  target: NodeID;
}
