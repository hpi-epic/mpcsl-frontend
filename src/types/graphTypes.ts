type NodeID = string | number;

export interface ID3Graph {
  nodes: ID3GraphNode[];
  links: ID3GraphLink[];
}

export interface ID3GraphNode {
  id: NodeID;
  x?: number;
  y?: number;
  fx?: number;
  fy?: number;
  vx?: number;
  vy?: number;
  isContext?: boolean;
}

export interface ID3GraphLink {
  source: NodeID;
  target: NodeID;
}

export interface IAPIGraphNode {
  id: number;
  name: string;
  result: number;
  result_id: number;
}

export interface IAPIGraphEdges {
  id: number;
  result: number;
  result_id: number;
  from_node_id: number;
  from_node: number;
  to_node_id: number;
  to_node: number;
  start_time: string;
}
