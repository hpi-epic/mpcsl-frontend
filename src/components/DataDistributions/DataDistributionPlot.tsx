import React from 'react';
import ContinousPlot from './ContinousPlot';
import { IAPIDistribution } from '../../types/types';
import CategoricalPlot from './CategoricalPlot';

interface IDataDistributionPlotProps {
  plotWidth: number;
  plotHeight: number;
  data: IAPIDistribution;
  selectable: boolean;
  onDataSelection?: (
    data:
      | { [bin: string]: number }
      | { startSelection: number; endSelection: number }
  ) => void;
  selection?: IDataSelection;
}

export type IDataSelection =
  | { [bin: string]: number }
  | { startSelection: number; endSelection: number };

class DataDistributionPlot extends React.Component<
  IDataDistributionPlotProps,
  {}
> {
  public render() {
    if (this.props.data.categorical) {
      return (
        <CategoricalPlot
          selectable={this.props.selectable}
          data={this.props.data!}
          plotWidth={this.props.plotWidth}
          plotHeight={this.props.plotHeight}
          onDataSelection={
            this.props.onDataSelection ? this.props.onDataSelection : undefined
          }
          selection={this.props.selection}
        />
      );
    } else {
      return (
        <ContinousPlot
          selectable={this.props.selectable}
          data={this.props.data!}
          plotWidth={this.props.plotWidth}
          plotHeight={this.props.plotHeight}
          onDataSelection={
            this.props.onDataSelection ? this.props.onDataSelection : undefined
          }
          selection={
            this.props.selection as {
              startSelection: number;
              endSelection: number;
            }
          }
        />
      );
    }
  }
}

export default DataDistributionPlot;
