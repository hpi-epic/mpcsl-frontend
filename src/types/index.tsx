import { IAPIGraphEdges, IAPIGraphNode, ID3GraphNode } from './graphTypes';
import { IndepenceTests } from '../constants/experiment';
import Graph from '../utils/graph';

export interface IStoreState {
  resultID?: string; // ID of current graph
  selectedGraph: Graph;
  nodes: ID3GraphNode[];
  doFreeze: boolean;
  availableNodes: IAPIGraphNode[];
}

export type StoreState = IStoreState | undefined;

export interface IIDClass {
  id: number;
}

export interface IObservationMatrix extends IIDClass {
  load_query: string;
  name: string;
  description?: string;
  data_source?: string;
  time_created?: string;
}

export type JobStatus = 'running' | 'done' | 'error' | 'cancelled' | 'waiting';

export enum BadgeStatus {
  running = 'processing',
  waiting = 'processing',
  done = 'success',
  error = 'error',
  cancelled = 'warning'
}

export interface IExperiment extends IIDClass {
  dataset_id: number;
  dataset?: number;
  name: string;
  description?: string;
  execution_time_statistics?: { [name: string]: number };
  algorithm_id: number;
  parameters: IParameters;
  last_job?: {
    id: number;
    experiment_id: number;
    start_time: string;
    pid: number;
    status: JobStatus;
    result?: {
      id: number;
      job_id: number;
      start_time: string;
      end_time: string;
      meta_results: any;
    };
  };
}

export interface ICreateExperiment {
  dataset_id: number;
  name: string;
  description: string;
  algorithm_id: number;
  parameters: { [name: string]: any };
}

interface IRequiredParameter {
  required?: boolean;
}

export interface INumberParameter extends IRequiredParameter {
  type: 'int' | 'float';
  minimum?: number;
  maximum?: number;
  default?: number;
}

export interface IEnumParameter extends IRequiredParameter {
  type: 'enum';
  values: string[];
  default?: string;
}

export type IParameter = INumberParameter | IEnumParameter;

export type IParameters = { [name: string]: IParameter };

export interface IAlgorithm extends IIDClass {
  name: string;
  script_filename: string;
  backend: string;
  description: string;
  valid_parameters: IParameters;
}

export interface IJob extends IIDClass {
  experiment_id: number;
  start_time: string;
  status: JobStatus;
  pid?: number;
  result?: {
    id: number;
    job_id: number;
    start_time: string;
    end_time: string;
    execution_time: number;
    dataset_loading_time: number;
    meta_results: any;
  };
}

export interface IAPIMetaResults {
  alpha: number;
  api_host: string;
  cores: number;
  dataset_id: string;
  fixed_edges: 'FALSE' | 'TRUE';
  fixed_graps: 'FALSE' | 'TRUE';
  help: boolean;
  independence_test: IndepenceTests;
  job_id: string;
}

export interface IAPIResult {
  nodes: IAPIGraphNode[];
  edges: IAPIGraphEdges[];
  meta_results: IAPIMetaResults;
  sepset: any[];
}

interface IAPIDistributionBase {
  dataset: {
    time_created: string;
    id: number;
    description: string | undefined;
    name: string;
    data_source: string | undefined;
    load_query: string | undefined;
  };
  node: {
    result: number;
    name: string;
    result_id: number;
    id: number;
  };
}

export interface IAPIDistributionContinous extends IAPIDistributionBase {
  categorical: false;
  bin_edges: number[];
  bins: number[];
}

interface ICategoricalBin {
  [bin: string]: number;
}

export interface IAPIDistributionCategorical extends IAPIDistributionBase {
  categorical: true;
  bins: ICategoricalBin;
}

export type IAPIDistribution =
  | IAPIDistributionCategorical
  | IAPIDistributionContinous;

export interface IAPINodeContext {
  edges: IAPIGraphEdges[];
  context_nodes: IAPIGraphNode[];
  main_node: IAPIGraphNode;
}

export enum GraphExportFormat {
  GEXF = 'GEXF',
  GraphML = 'GraphML',
  GML = 'GML'
}

export interface IAPIConfounders {
  confounders: string[][];
}
