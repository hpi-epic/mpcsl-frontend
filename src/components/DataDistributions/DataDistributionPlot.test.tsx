import React from 'react';
import renderer from 'react-test-renderer';
import DataDistributionPlot from './DataDistributionPlot';
import { IAPIDistributionCategorical } from '../../types/types';

describe('<DataDistributinoPlot />', () => {
  it('works for categorical data', () => {
    const categoricalAPIDataDistribution = {
      categorical: true,
      bins: {
        1: 10
      }
    };
    const tree = renderer
      .create(
        <DataDistributionPlot
          plotHeight={100}
          plotWidth={100}
          selectable={true}
          data={
            (categoricalAPIDataDistribution as unknown) as IAPIDistributionCategorical
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('works for continous data', () => {
    const categoricalAPIDataDistribution = {
      categorical: false,
      bin_edges: [1, 2, 3, 4],
      bins: [10, 2, 5, 5]
    };
    const tree = renderer
      .create(
        <DataDistributionPlot
          selectable={true}
          plotHeight={100}
          plotWidth={100}
          data={
            (categoricalAPIDataDistribution as unknown) as IAPIDistributionCategorical
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
