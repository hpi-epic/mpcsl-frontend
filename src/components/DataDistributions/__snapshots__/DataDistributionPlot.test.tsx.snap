// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataDistributinoPlot /> works for categorical data 1`] = `
<div>
  <div
    className="rv-xy-plot "
    style={
      Object {
        "height": "100px",
        "width": "100px",
      }
    }
  >
    <svg
      className="rv-xy-plot__inner"
      height={100}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      onWheel={[Function]}
      width={100}
    >
      <g
        className="rv-xy-plot__grid-lines"
        transform="translate(40,10)"
      >
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={25}
          x2={25}
          y1={0}
          y2={50}
        />
      </g>
      <g
        className="rv-xy-plot__grid-lines"
        transform="translate(40,10)"
      >
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={50}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={40}
          y2={40}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={30}
          y2={30}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={20}
          y2={20}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={10}
          y2={10}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={0}
          y2={0}
        />
      </g>
      <g
        className="rv-xy-plot__axis rv-xy-plot__axis--horizontal "
        style={Object {}}
        transform="translate(40,60)"
      >
        <line
          className="rv-xy-plot__axis__line"
          style={Object {}}
          x1={0}
          x2={50}
          y1={0}
          y2={0}
        />
        <g
          className="rv-xy-plot__axis__ticks"
          transform="translate(0, 0)"
        >
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(25, 0)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={0}
              x2={0}
              y1={-6}
              y2={6}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.72em"
              style={Object {}}
              textAnchor="middle"
              transform="translate(0, 14)"
            >
              1
            </text>
          </g>
        </g>
      </g>
      <g
        className="rv-xy-plot__axis rv-xy-plot__axis--vertical "
        style={Object {}}
        transform="translate(0,10)"
      >
        <line
          className="rv-xy-plot__axis__line"
          style={Object {}}
          x1={40}
          x2={40}
          y1={0}
          y2={50}
        />
        <g
          className="rv-xy-plot__axis__ticks"
          transform="translate(40, 0)"
        >
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 50)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              0%
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 40)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              20%
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 30)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              40%
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 20)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              60%
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 10)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              80%
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 0)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              100%
            </text>
          </g>
        </g>
      </g>
      <g
        className="rv-xy-plot__series rv-xy-plot__series--bar "
        transform="translate(40,10)"
      >
        <rect
          height={50}
          onClick={[Function]}
          onContextMenu={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "fill": "#12939A",
              "opacity": 0.8,
              "stroke": "#fff",
            }
          }
          width={42.5}
          x={3.75}
          y={0}
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`<DataDistributinoPlot /> works for continous data 1`] = `
<div>
  <div
    className="rv-xy-plot "
    style={
      Object {
        "height": "100px",
        "width": "100px",
      }
    }
  >
    <svg
      className="rv-xy-plot__inner"
      height={100}
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      onWheel={[Function]}
      width={100}
    >
      <g
        className="rv-xy-plot__grid-lines"
        transform="translate(40,10)"
      >
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={0}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={8.333333333333332}
          x2={8.333333333333332}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={16.666666666666664}
          x2={16.666666666666664}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={25}
          x2={25}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={33.33333333333333}
          x2={33.33333333333333}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={41.66666666666667}
          x2={41.66666666666667}
          y1={0}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={50}
          x2={50}
          y1={0}
          y2={50}
        />
      </g>
      <g
        className="rv-xy-plot__grid-lines"
        transform="translate(40,10)"
      >
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={50}
          y2={50}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={40.90909090909091}
          y2={40.90909090909091}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={31.818181818181817}
          y2={31.818181818181817}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={22.72727272727273}
          y2={22.72727272727273}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={13.636363636363633}
          y2={13.636363636363633}
        />
        <line
          className="rv-xy-plot__grid-lines__line"
          x1={0}
          x2={50}
          y1={4.545454545454547}
          y2={4.545454545454547}
        />
      </g>
      <g
        className="rv-xy-plot__axis rv-xy-plot__axis--horizontal "
        style={Object {}}
        transform="translate(40,60)"
      >
        <line
          className="rv-xy-plot__axis__line"
          style={Object {}}
          x1={0}
          x2={50}
          y1={0}
          y2={0}
        />
        <g
          className="rv-xy-plot__axis__ticks"
          transform="translate(0, 0)"
        >
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(8.333333333333332, 0)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={0}
              x2={0}
              y1={-6}
              y2={6}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0"
              style={Object {}}
              textAnchor="end"
              transform="translate(0, 14) rotate(-45)"
            >
              1.5
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(25, 0)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={0}
              x2={0}
              y1={-6}
              y2={6}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0"
              style={Object {}}
              textAnchor="end"
              transform="translate(0, 14) rotate(-45)"
            >
              2.5
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(41.66666666666667, 0)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={0}
              x2={0}
              y1={-6}
              y2={6}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0"
              style={Object {}}
              textAnchor="end"
              transform="translate(0, 14) rotate(-45)"
            >
              3.5
            </text>
          </g>
        </g>
      </g>
      <g
        className="rv-xy-plot__axis rv-xy-plot__axis--vertical "
        style={Object {}}
        transform="translate(0,10)"
      >
        <line
          className="rv-xy-plot__axis__line"
          style={Object {}}
          x1={40}
          x2={40}
          y1={0}
          y2={50}
        />
        <g
          className="rv-xy-plot__axis__ticks"
          transform="translate(40, 0)"
        >
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 50)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              0
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 40.90909090909091)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              2
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 31.818181818181817)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              4
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 22.72727272727273)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              6
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 13.636363636363633)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              8
            </text>
          </g>
          <g
            className="rv-xy-plot__axis__tick"
            style={Object {}}
            transform="translate(0, 4.545454545454547)"
          >
            <line
              className="rv-xy-plot__axis__tick__line"
              style={Object {}}
              x1={6}
              x2={-6}
              y1={0}
              y2={0}
            />
            <text
              className="rv-xy-plot__axis__tick__text"
              dy="0.32em"
              style={Object {}}
              textAnchor="end"
              transform="translate(-14, 0)"
            >
              10
            </text>
          </g>
        </g>
      </g>
      <g
        className="rv-xy-plot__series rv-xy-plot__series--rect "
        transform="translate(40,10)"
      >
        <rect
          height={45.45454545454545}
          onClick={[Function]}
          onContextMenu={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "fill": "#12939A",
              "opacity": 0.8,
              "stroke": "#fff",
            }
          }
          width={16.666666666666664}
          x={0}
          y={4.545454545454547}
        />
        <rect
          height={9.090909090909093}
          onClick={[Function]}
          onContextMenu={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "fill": "#12939A",
              "opacity": 0.8,
              "stroke": "#fff",
            }
          }
          width={16.666666666666664}
          x={16.666666666666664}
          y={40.90909090909091}
        />
        <rect
          height={22.727272727272727}
          onClick={[Function]}
          onContextMenu={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "fill": "#12939A",
              "opacity": 0.8,
              "stroke": "#fff",
            }
          }
          width={16.66666666666667}
          x={33.33333333333333}
          y={27.272727272727273}
        />
        <rect
          height={22.727272727272727}
          onClick={[Function]}
          onContextMenu={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "fill": "#12939A",
              "opacity": 0.8,
              "stroke": "#fff",
            }
          }
          width={NaN}
          x={50}
          y={27.272727272727273}
        />
      </g>
    </svg>
  </div>
</div>
`;
